### PR TITLE
Implement fillager api

### PR DESCRIPTION
**Pull request by sjoblomjsqueed**
Using OpenAPI and new endpoints to access Soknadsfillager.

The PR consists of a few commits, it is probably a good idea to look at these one by one.

### CODE REVIEW

File: arkiverer/src/main/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageService.kt [hunk 1]
```diff
@@ -32,15 +36,16 @@ class FilestorageService(@Qualifier("basicWebClient") private val webClient: Web
...
 
-	override fun getFilesFromFilestorage(key: String, data: Soknadarkivschema): List<FilElementDto> {
+	override fun getFilesFromFilestorage(key: String, data: Soknadarkivschema): List<FileData> {
 		val timer = metrics.filestorageGetLatencyStart()
 		try {
 			val fileIds = getFileIds(data)
 			logger.info("$key: Getting files with ids: '$fileIds'")
 
 			val files = getFiles(fileIds)
+				.map { FileData(it.uuid, it.fil, OffsetDateTime.now(ZoneOffset.UTC)) }
```
Comment (sjoblomjsqueed) [inline arkiverer/src/main/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageService.kt]: Soknadsfillager still returns `FilElementDto`s, but here we convert it to the new data type that will be used instead.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/ApplicationTests.kt [hunk 2]
```diff
@@ -478,7 +478,7 @@ class ApplicationTests {
 
 
 	private fun verifyDeleteRequestsToFilestorage(expectedCount: Int) {
-		verifyMockedDeleteRequests(expectedCount, appConfiguration.config.filestorageUrl.replace("?", "\\?") + ".*")
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/ApplicationTests.kt:481]: The url used to be `/files?ids=1,2,3`, but is now `/files/1,2,3`. No more need to escape `?` in tests

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageServiceTests.kt [hunk 3]
```diff
@@ -119,6 +119,7 @@ class FilestorageServiceTests {
 		verifyMockedGetRequests(1, makeUrl(fileIdsAndResponses.drop(filesInOneRequestToFilestorage * 2).take(1)))
 	}
 
+	@Disabled
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageServiceTests.kt]: Test intentionally broken. Will be fixed in next commit

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageServiceTests.kt [hunk 4]
```diff
@@ -119,13 +119,9 @@ class FilestorageServiceTests {
 		verifyMockedGetRequests(1, makeUrl(fileIdsAndResponses.drop(filesInOneRequestToFilestorage * 2).take(1)))
 	}
 
-	@Disabled
 	@Test
-	fun `getFilesFromFilestorage - Asking for 3 files - Only 2 is returned - will throw exception`() {
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/service/fileservice/FilestorageServiceTests.kt:123]: Old test case no longer valid - Soknadsfillager cannot respond with a different amount of files than what was asked for. Updating with new test that captures new behaviour.

**Review by nils-arne — APPROVED** [review]

**Pull request merged**